src/libdredd/src/mutate_visitor.cc

```diff
@@ -63,6 +64,25 @@ bool MutateVisitor::IsTypeSupported(const clang::QualType qual_type) {
          (builtin_type->isInteger() || builtin_type->isFloatingPoint());
 }
 
+void MutateVisitor::UpdateStartLocationOfFirstFunctionInSourceFile() {
+  for (int index = static_cast<int>(enclosing_decls_.size()) - 1; index >= 0;
+       index--) {
+    const auto* decl = enclosing_decls_[static_cast<size_t>(index)];
+    if (llvm::dyn_cast<clang::FunctionDecl>(decl) != nullptr) {
+      const clang::BeforeThanCompare<clang::SourceLocation> comparator(
+          compiler_instance_->getSourceManager());
+      auto source_range_in_main_file = GetSourceRangeInMainFile(
+          compiler_instance_->getPreprocessor(), *enclosing_decls_[0]);
+      if (start_location_of_first_function_in_source_file_.isInvalid() ||
+          comparator(source_range_in_main_file.getBegin(),
+                     start_location_of_first_function_in_source_file_)) {
+        start_location_of_first_function_in_source_file_ =
+            source_range_in_main_file.getBegin();
+      }
+    }
+  }
```

Comment:
Use `IsInFunction()` rather than repeating.

Reply:
@JamesLee-Jones I didn't fully understand this comment and am not sure whether this is a remaining TODO, or something to be addressed in a follow up PR. If there is a way to simplify this then please go ahead.